### PR TITLE
Initial stack navigator

### PR DESCRIPTION
**Pull request by czystyl**
Related #6 

TODO: 
- [x] Use Route Variants to define `initialRouteName`
- [ ] Accept `make` and `component` in the `screen` property inside `routeConfiguration`
- [ ] Pass navigation to the first level components

### CODE REVIEW

File: example/src/App.re [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-let reactClass =
-  BsReactNavigation.Navigation.createStackNavigator(
-    Home.reactClass,
-    Detail.reactClass,
-  );
+let navigationConfig =
+  StackNavigator.createNavigationConfig(~initialRouteName="Home");
+
+let routes = [
+  ("Home", StackNavigator.createRoute(~screen=() => <Home />)),
```
Comment (baransu) [inline example/src/App.re]: The cool thing is we can use `[@bs.deriving jsConverter]` on polymorphic variant and have the route type which can be used in `navigate`

Comment (czystyl) [reply example/src/App.re]: I was playing with functors to solve a problem with the nested navigator. 
Still, need to resolve some problems but going forward. Will be public that concept soon :)

File: example/src/App.re [hunk 2]
```diff
@@ -5,32 +5,17 @@ module Config = {
...
-        "Details",
-        {
-          screen:
-            ReasonReact.wrapReasonForJs(
-              ~component=Details.component,
-              Details.make,
-            ),
-        },
-      )
+    | Home => ("Home", {screen: Home.reactComponent})
```
Comment (baransu) [inline example/src/App.re]: I really like that 😃

Comment (czystyl) [reply example/src/App.re]: In the next version we can think about to remove the string from this Variant but for now, it's not that bad :P

File: example/src/App.re [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+  open StackNavigator;
 
-/* let reactClass = nav(Home.reactClass, Detail.reactClass); */
-/* let reactClass = Detail.reactClass; */
\ No newline at end of file
+  type routes =
+    | Home
+    | Details;
+
+  let navigatorConfig = {initialRouteName: "Home"};
```
Comment (czystyl) [inline example/src/App.re]: Try to use variant's here:

```
  let navigatorConfig = {initialRouteName: Home};
```

File: example/src/App.re [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+  open StackNavigator;
 
-/* let reactClass = nav(Home.reactClass, Detail.reactClass); */
-/* let reactClass = Detail.reactClass; */
\ No newline at end of file
+  type routes =
+    | Home
+    | Details;
+
+  let navigatorConfig = {initialRouteName: Details};
```
Comment (czystyl) [inline example/src/App.re]: @baransu Uff going forward but with small steps :P

**Pull request merged**